dynatrace_event/tests/test_dynatrace_event.py

```diff
@@ -0,0 +1,261 @@
+# (C) StackState 2020
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from stackstate_checks.dynatrace_event import DynatraceEventCheck, util, EventLimitReachedException
+from stackstate_checks.base.stubs import aggregator
+
+import pytest
+import unittest
+import os
+import mock
+import yaml
+import shutil
+import json
+
+
+def read_data(filename):
+    with open("./tests/samples/" + filename, "r") as f:
+        data = yaml.safe_load(f)
+        return data
+
+
+def sort_events_data(events):
+    events = [json.dumps(event, sort_keys=True) for event in events]
+    return events
+
+
+@pytest.mark.usefixtures("instance")
+class TestDynatraceEventCheck(unittest.TestCase):
+    """Basic Test for Dynatrace integration."""
+    CHECK_NAME = 'dynatrace'
+    SERVICE_CHECK_NAME = "dynatrace"
+
+    def setUp(self):
+        """
+        Initialize and setup the check
+        """
+        config = {}
+        self.check = DynatraceEventCheck(self.CHECK_NAME, config, instances=[self.instance])
+
+        # override the path as the conf.d folder doesn't exist here
+        util.DYNATRACE_STATE_FILE = os.getcwd() + "/dynatrace_event.d/dynatrace_event_state.pickle"
+        if not os.path.exists(os.getcwd()+"/dynatrace_event.d/"):
+            os.mkdir(os.getcwd() + "/dynatrace_event.d/")
+
+        # this is needed because the topology retains data across tests
+        aggregator.reset()
+
+    def tearDown(self):
+        """
+        Destroy the environment
+        """
+        if os.path.exists(os.getcwd()+"/dynatrace_event.d/"):
+            print("deleting the dir")
+            shutil.rmtree(os.getcwd()+"/dynatrace_event.d/")
+
+    def test_check_for_empty_events(self):
+        """
+        Testing Dynatrace event check should not produce any events
+        """
+        self.check.get_json_response = mock.MagicMock()
+        self.check.get_json_response.return_value = []
+        self.check.url = self.instance.get('url')
+
+        self.check.check(self.instance)
+
+        events = aggregator.events
+        self.assertEqual(len(events), 0)
+
+    def test_check_for_events_process_limit(self):
+        """
+        Testing Dynatrace should respect `events_process_limit` config and just produce those number of events
+        """
+        self.check.get_json_response = mock.MagicMock(return_value=read_data("full_events.json"))
+        self.check._current_time_seconds = mock.MagicMock(return_value=1602685050)
+        self.check.url = self.instance.get('url')
+        self.instance["events_process_limit"] = 10
+
+        self.check.check(self.instance)
+
+        events = sort_events_data(aggregator.events)
+        self.assertEqual(len(events), self.instance.get("events_process_limit"))
+        # make sure we have only 2 open events in state
```

Comment:
We are not storing at the end of the run but while processing we store for each `entityId` either `CLOSED` or `OPEN`. then send `CLEAR` for closed `entityId` and for open `entityId` aggregate. Isn't it better to show green components rather than `grey` because anyway if any events come for those green one again then we process those?